### ISSUE TITLE
php-pdlib-1.1.0/lib/php/extensions/pdlib.so: undefined symbol: cblas_strsm when nextcloud-occ is used

### ISSUE THREAD

**Issue by marmolak** [NixOS/nixpkgs#318913]
### Describe the bug
Module not found warning when `nextcloud-occ` was run.
Seems like face recognization doesn't work.

### Steps To Reproduce
1. Set: 
  ```
services.nextcloud = { 
... 
   phpExtraExtensions = all: [ all.pdlib all.bz2 ];
```
2. install recognize 7.0.0 (appstore, nextcloud configuration)
3. run: `nextcloud-occ files:scan-app-data` 

Result:
```
Warning: PHP Startup: Unable to load dynamic library '/nix/store/fwjzsld9njr8l50fj15dv5zmnvd7xf98-php-pdlib-1.1.0/lib/php/extensions/pdlib.so' (tried: /nix/store/fwjzsld9njr8l50fj15dv5zmnvd7xf98-php-pdlib-1.1.0/lib/php/extensions/pdlib.so (/nix/store/fwjzsld9njr8l50fj15dv5zmnvd7xf98-php-pdlib-1.1.0/lib/php/extensions/pdlib.so: undefined symbol: cblas_strsm), /nix/store/m371rkz0wiplvfdiyigys28dskgy0mq6-php-8.3.8/lib/php/extensions//nix/store/fwjzsld9njr8l50fj15dv5zmnvd7xf98-php-pdlib-1.1.0/lib/php/extensions/pdlib.so.so (/nix/store/m371rkz0wiplvfdiyigys28dskgy0mq6-php-8.3.8/lib/php/extensions//nix/store/fwjzsld9njr8l50fj15dv5zmnvd7xf98-php-pdlib-1.1.0/lib/php/extensions/pdlib.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
Scanning AppData for files
```
and seems like face recognization doesn't work

### Expected behavior
No unknown symbol.

### Screenshots
None

### Additional context
None

### Notify maintainers

@drupol 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.8.12, NixOS, 24.05 (Uakari), 24.05.20240609.2819fff`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(rhack): `"nixos-23.11"`
 - channels(root): `"nixos-23.05"`
 - nixpkgs: `/nix/store/giq1qbcwfx5gq0g9jf6id0b3m55a860j-source`
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by drupol** [comment]
I've been debugging this... but I don't think the issue comes from Nix.

```
❯ nix build --expr '(builtins.getFlake "/home/pol/Code/NixOS/nixpkgs").legacyPackages.${builtins.currentSystem}.php83.extensions.pdlib' --impure -L
❯ ll
.rw-r--r--  37k pol 11 jun 08:10 CONTRIBUTING.md
.rw-r--r-- 1,1k pol 11 jun 08:06 COPYING
.rw-r--r--  971 pol 12 mar 10:38 default.nix
drwxr-xr-x    - pol 11 jun 08:10 doc
.rw-r--r-- 4,6k pol 11 jun 08:10 flake.nix
drwxr-xr-x    - pol 11 jun 08:10 lib
drwxr-xr-x    - pol 11 jun 08:10 maintainers
drwxr-xr-x    - pol 11 jun 08:10 nixos
drwxr-xr-x    - pol 11 jun 08:10 pkgs
.rw-r--r-- 5,3k pol 11 jun 08:06 README.md
lrwxrwxrwx    - pol 11 jun 08:36 result -> /nix/store/kql6nif5wycy7p73flw2ziy1c21bw227-php-pdlib-1.1.0
❯ ldd ./result/lib/php/extensions/pdlib.so 
        linux-vdso.so.1 (0x00007f5ab2908000)
        libstdc++.so.6 => /nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib/lib/../lib64/libstdc++.so.6 (0x00007f5ab2400000)
        libm.so.6 => /nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52/lib/libm.so.6 (0x00007f5ab2663000)
        libc.so.6 => /nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52/lib/libc.so.6 (0x00007f5ab2213000)
        libgcc_s.so.1 => /nix/store/xvzz97yk73hw03v5dhhz3j47ggwf1yq1-gcc-13.2.0-lib/lib/../lib64/libgcc_s.so.1 (0x00007f5ab21ee000)
        /nix/store/k7zgvzp2r31zkg9xqgjim7mbknryv6bs-glibc-2.39-52/lib64/ld-linux-x86-64.so.2 (0x00007f5ab290a000)
❯
```

**Comment by marmolak** [comment]
Thank you very much. I'm going to use l/strace to figure out from where nextcloud-occ has been calling the library with the missing symbol.

**Comment by CaptainJawZ** [comment]
> Thank you very much. I'm going to use l/strace to figure out from where nextcloud-occ has been calling the library with the missing symbol.

could you figure it out? having the same issue.

**Comment by marmolak** [comment]
Hi. No progress.
According to strace, library with missing symbol is loaded within nextcloud-occ (which is bash script for php).
And then I'm lost because I'm not sure how to walk how libraries (and php extensions) were loaded.

**Comment by benneti** [comment]
I don't know whether this information helps at all, but for me this problem started only after upgrading to nixos 24.05 (I am however not sure whether in the same upgrade I did upgrade facerecognition itself).
As the only change in nixos directly obvious to me to the nextcloud setup itself was a change in the default php version, I also checked whether downgrading only php helps (which it does not).
Also manually setting the LD_LIBRARY_PATH in the phpEnv did not seem to help either (but it might not be the best place to override libraries anyway).

**Comment by CaptainJawZ** [comment]
Same thing, I didnt virtually changed anything regarding nextcloud when upgrading from 23.11 to 24.05 and the bug appeared after the upgrade, Ive tried many things, like changing the compile flags, and checking the file that loads php.ini files, it all seems to be working in order, I also tried what you did, when it comes to downgrading php, but I don't think it's an issue with php, I checked the git history, and tried undoing this nix commit with a flake input https://github.com/NixOS/nixpkgs/commit/c759efa5e7f825913f9a69ef20f025f50f56dc4d and didn't helped either, so it may be a problem with nextcloud, but I am clueless on how to look into that, nextcloud has also shown me some other bugs (minor tho) as of lately

**Comment by benneti** [comment]
I actually do not think it is a problem  with nextcloud in particular.
If on my laptop I run:
`nix-shell -p 'pkgs.php.buildEnv { extensions = {enabled, all}: enabled ++ (with all; [ pdlib ]); }' `
I also get the 
`Warning: PHP Startup: Unable to load dynamic library '/nix/store/da0s22pa0gq2zdvk64461hvyn3lacclq-php-pdlib-1.1.0/lib/php/extensions/pdlib.so' (tried: /nix/store/da0s22pa0gq2zdvk64461hvyn3lacclq-php-pdlib-1.1.0/lib/php/extensions/pdlib.so (/nix/store/da0s22pa0gq2zdvk64461hvyn3lacclq-php-pdlib-1.1.0/lib/php/extensions/pdlib.so: undefined symbol: cblas_strsm), /nix/store/nlb71gfszca6gc2pk3xg581r3rs36kw8-php-8.2.23/lib/php/extensions//nix/store/da0s22pa0gq2zdvk64461hvyn3lacclq-php-pdlib-1.1.0/lib/php/extensions/pdlib.so.so (/nix/store/nlb71gfszca6gc2pk3xg581r3rs36kw8-php-8.2.23/lib/php/extensions//nix/store/da0s22pa0gq2zdvk64461hvyn3lacclq-php-pdlib-1.1.0/lib/php/extensions/pdlib.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0`

**Comment by turion** [comment]
CC @aanderse

**Comment by turion** [comment]
I cannot reproduce on current nixos-unstable, are you sure this still occurs? On an older version I could reproduce.

**Comment by turion** [comment]
(I'm currently on 18536bf04cd71abd345f9579158841376fdd0c5a)

**Comment by benneti** [comment]
on nixos stable  'github:nixos/nixpkgs/cd3e8833d70618c4eea8df06f95b364b016d4950' (2024-10-26) it is still occuring for me. However, I can also confirm that it seems to be fixed on unstable.

**Comment by CaptainJawZ** [comment]
I too confirm, that it is fixed on unstable.

**Comment by turion** [comment]
I tried to bisect, I didn't get the exact commit. But probably this was fixed in 3a8211a3aeef.

**Comment by benneti** [comment]
Thanks for the work, I can indeed confirm that also on nixos-24.05 when overriding pdlib with the version from from unstable it works, i.e., the issue seems to not lie within php.
I am not sure whether it is worth it this late in the cycle, but maybe the newer pdlib would be nice to be backportet to nixos-24.05, what do you think?
